checker/src/org/checkerframework/checker/upperbound/UpperBoundVisitor.java

```diff
@@ -0,0 +1,131 @@
+package org.checkerframework.checker.upperbound;
+
+import com.sun.source.tree.ArrayAccessTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.ArrayList;
+import java.util.List;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.upperbound.qual.LTLengthOf;
+import org.checkerframework.checker.upperbound.qual.LTOMLengthOf;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.framework.source.Result;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.TreeUtils;
+
+/** Warns about array accesses that could be too high. */
+public class UpperBoundVisitor extends BaseTypeVisitor<UpperBoundAnnotatedTypeFactory> {
+
+    protected final List<ExecutableElement> IndexFirstArgListMethods;
```

Comment:
I agree that this code can be removed once the JDK is completely annotated for the whole index checker.
However, I'm unsure of how this code relates to [kelloggm/checker-framework#14](https://github.com/kelloggm/checker-framework/issues/14). My planned solution to that issue is to add a refinement rule to UpperBoundTransfer that, whenever it encounters a compile time constant, it checks the MinLenStore at that program point and looks to see if the value is less than the minlen of that array; if it is, it would add an LTL of that array. The only reason that's not implemented yet is that I don't know how to get a MinLenStore in UpperBoundTransfer.